### PR TITLE
adds keyboard input control

### PR DESCRIPTION
**Pull request by VODU**
- adds cpp-terminal dep as submodule
- adds terminal keyboard input for start/stop and user responding to the test.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "ext/rtaudio"]
 	path = ext/rtaudio
 	url = git@github.com:thestk/rtaudio.git
+[submodule "ext\\cpp-terminal"]
+	path = ext\\cpp-terminal
```
Comment (binarybleeps) [inline .gitmodules:5]: Different path formatting between the submodules -- would be nicer to use coherent formatting.

File: main.cpp [hunk 2]
```diff
@@ -106,15 +174,18 @@ int main(int /*argc*/, char /**argv[]*/)
...
-        catch (RtAudioError &e) {
+        catch (RtAudioError &e)
+        {
             std::cerr << e.getMessage() << "\n";
             return false;
-        }		
+        }
     }
     return true;
 }
```
Comment (binarybleeps) [inline main.cpp:191]: newline..

File: main.cpp [hunk 3]
```diff
@@ -12,92 +16,156 @@ std::shared_ptr<HearingTestEngine> hte;
...
     stream.nChannels = NUMCHANNELS;
-    
+
     // setting up hte
     hte = HearingTestEngine::create();
-    std::string hte_version = hte->version.description();
-    std::cout << "\n=== HTE: " << hte->version.description() << " ===" << "\n";
-    
+
     EngineStatusType curr_eng_status = EngineStatusType::Unknown;
```
Comment (binarybleeps) [inline main.cpp:47]: `curr_eng_status` in snake case whilst most of the code here is in camel case.

File: .gitmodules (hunk 1 above)
Comment (VODU) [reply .gitmodules:5]: ups..windows 🤦‍♂️

Comment (VODU) [reply .gitmodules]: fixed

File: main.cpp (hunk 2 above)
Comment (VODU) [reply main.cpp]: fixed

File: main.cpp (hunk 3 above)
Comment (VODU) [reply main.cpp]: fixed

**Review by binarybleeps — APPROVED** [review]

**Pull request merged**